compiler/rustc_resolve/src/diagnostics.rs

```diff
@@ -2623,7 +2623,13 @@ impl<'ra, 'tcx> Resolver<'ra, 'tcx> {
         };
 
         for &StrippedCfgItem { parent_module, ident, ref cfg } in symbols {
-            if parent_module != module || ident.name != *segment {
+            let comes_from_same_module = parent_module == module
+                || self.module_map.get(&parent_module).is_some_and(|m| {
```

Comment:
Maybe also makes sense to evaluate the expensive condition last, after both `ident.name != *segment` and `parent_module == module`, but it's a purely diagnostic branch so it's not very important.